src/tangle.ts

```diff
@@ -51,17 +64,59 @@ export class Client<T> {
         return this._inbound.value.transient;
     }
 
+    /**
+     * notify bus of new sandbox
+     */
+    public notify() {
+        this._context.clients.set(this.id, true);
+        this._notifer.next(this._context);
+    }
+
+    /**
+     * collect and hold information about connected sandboxes
+     */
+    private _registerContext() : Observable<Context> {
+        const context$ = this._inbound.pipe(
+            pluck('context'),
+            filter(Boolean),
+            scan((acc, curr) => {
+                if (typeof curr?.clients?.forEach === 'function') {
+                    curr?.clients?.forEach((function(value, key) {
+                        acc.clients.set(key, value);
+                    }));
+                }
+                return acc;
+            }, this._context),
+            share()
+        );
+        context$.subscribe(ctx => {
+            this._context.clients = ctx.clients;
+        });
+
+        return context$;
+    }
+
+    /**
+     * returns promise that fires when all sandboxes have connected
+     */
+    public readyPromise(): Promise<Context>{
```

Comment:
That said, we already have `registerPromise` so we at least align with the current interface, though that one has a non-promise alternative.

Reply:
> That said, we already have `registerPromise` so we at least align with the current interface, though that one has a non-promise alternative.

I chose `readyPromise` (as a placeholder) for the exact reason to provide consistency with `registerPromise`. I do agree that `readyPromise` is a fairly odd name. My mind also went to `whenReady` or `onReady`. `on` is kinda reserved for the emitter interface so `whenReady` is probably best. `allReady` perhaps, since it waits for all provided sandboxes to be resolved/attached?

Reply:
> My mind also went to whenReady

Let's settle on it then.